File: programmers/pg_lv3_여행경로.js
function solution(tickets) {
  const result = [];
  const ticketsMap = new Map();
  tickets.forEach((ele) => {
    if (ticketsMap.has(ele[0])) ticketsMap.get(ele[0]).push(ele[1]);
    else ticketsMap.set(ele[0], [ele[1]]);
  });

  for (const key of ticketsMap.keys()) {
    ticketsMap.get(key).sort((a, b) => (a > b ? -1 : 1));
  }

  const stack = ["ICN"];
  while (stack.length) {
    const airport = stack[stack.length - 1];

    if (!ticketsMap.has(airport) || ticketsMap.get(airport).length <= 0)
      result.push(stack.pop());
    else stack.push(ticketsMap.get(airport).pop());
  }

  return result.reverse();
}

File: python/programmers/pg_lv3_여행경로.py
def solution(tickets):
    result = []
    airportDict = dict()
    for (a, b) in tickets:
        if a in airportDict: airportDict.get(a).append(b)
        else: airportDict[a] = [b]
        
    for (key, value) in airportDict.items():
        value.sort(reverse = True)
        
    stack = ["ICN"]
    while stack:
        if stack[-1] not in airportDict or len(airportDict.get(stack[-1])) == 0: result.append(stack.pop())
        else: stack.append(airportDict.get(stack[-1]).pop())
    
    result.reverse()
    return result